### PR TITLE
Add option to add custom material to GPUPicker

### PR DESCRIPTION
**Pull request by vinhui**
In response to my own post https://forum.babylonjs.com/t/new-feature-gpu-picking/51106/54?u=vinhui

This PR allows the user to pass a custom material to the `GPUPicker.setPickingList` method.

I tested with a bunch of other existing playgrounds to see if I broke anything.
I created a playground that shows how to use this PR `#4RKT66#4`. In this PR, there's a gui with a transparent background. The shader then ignores any transparent pixels when picking. There's a faint black border around the GUI that shows the border of the plane.

Planes with transparent textures are the obvious use-case for this PR (sprites for example), but there are probably other use-cases.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15350/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15350/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15350/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15350/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15350/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15350/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15350/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15350/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15350/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15350/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15350/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by RaananW** [comment]
There seems to be a transparency discrepancy (no pun intended :-) ) between the previous implementation and this one.

**Comment by deltakosh** [comment]
I also have a concern regarding perf of checking the presence of obj.mesh on ALL calls.

**Comment by vinhui** [comment]
> I also have a concern regarding perf of checking the presence of obj.mesh on ALL calls.

I have an idea on how to fix that, it will probably streamline the code in general

**Comment by deltakosh** [comment]
ok excellent! Thanks a lot!

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15350/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Collisions/gpuPicker.ts [hunk 1]
```diff
@@ -366,7 +389,7 @@ export class GPUPicker {
         // Cleaning up
         this._pickingTexure?.dispose();
         this._pickingTexure = null;
-        this._renderMaterial?.dispose();
-        this._renderMaterial = null;
+        this._defaultRenderMaterial?.dispose();
```
Comment (deltakosh) [inline packages/dev/core/src/Collisions/gpuPicker.ts:392]: let's make sure to also wipe the map and we are all good!

Comment (vinhui) [reply packages/dev/core/src/Collisions/gpuPicker.ts:392]: Good point, I'll do that tomorrow

Comment (vinhui) [reply packages/dev/core/src/Collisions/gpuPicker.ts:392]: Looking at the code again, that already happens with the `this.setPickingList(null);` call. Or would you prefer it being set to `null`?

**Comment by matthargett** [comment]
cc @hmallen99

File: packages/dev/core/src/Collisions/gpuPicker.ts (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/Collisions/gpuPicker.ts:392]: Correct! we are all good :)

**Review by deltakosh — APPROVED** [review]

**Pull request merged**